sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -2111,19 +2112,24 @@ struct SptrsvWrap {
       for (int i = 0; i < nstreams; i++) {
         // Only if stream i-th still has this level
         if (lvl < nlevels_v[i]) {
-          size_type lvl_nodes = hnodes_per_level_v[i](lvl);
+          const size_type lvl_nodes = hnodes_per_level_v[i](lvl);
+          const auto block_size     = thandle_v[i]->get_block_size();
+          const auto block_enabled  = thandle_v[i]->is_block_enabled();
+          KK_REQUIRE(block_enabled == BlockEnabled);
           if (lvl_nodes != 0) {
             if (thandle_v[i]->get_algorithm() == KokkosSparse::Experimental::SPTRSVAlgorithm::SEQLVLSCHD_RP) {
               Kokkos::parallel_for("parfor_fixed_lvl",
                                    range_policy(execspace_v[i], node_count_v[i], node_count_v[i] + lvl_nodes),
-                                   RPPointFunctor(row_map_v[i], entries_v[i], values_v[i], lhs_v[i], rhs_v[i],
-                                                  nodes_grouped_by_level_v[i]));
+                                   RPFunctor(row_map_v[i], entries_v[i], values_v[i], lhs_v[i], rhs_v[i],
+                                             nodes_grouped_by_level_v[i], block_size));
             } else if (thandle_v[i]->get_algorithm() == KokkosSparse::Experimental::SPTRSVAlgorithm::SEQLVLSCHD_TP1) {
               int team_size = thandle_v[i]->get_team_size();
               auto tp       = team_size == -1 ? team_policy(execspace_v[i], lvl_nodes, Kokkos::AUTO)
                                               : team_policy(execspace_v[i], lvl_nodes, team_size);
-              TPPointFunctor tstf(row_map_v[i], entries_v[i], values_v[i], lhs_v[i], rhs_v[i],
-                                  nodes_grouped_by_level_v[i], node_count_v[i]);
+              TPFunctor tstf(row_map_v[i], entries_v[i], values_v[i], lhs_v[i], rhs_v[i], nodes_grouped_by_level_v[i],
+                             node_count_v[i], block_size);
+              const int scratch_size = TPFunctor::SBlock::shmem_size(block_size, block_size);
+              tp                     = tp.set_scratch_size(0, Kokkos::PerTeam(scratch_size));
```

Comment:
@jgfouca Am I missing something here? Why do we call ```set_scratch_size``` and assign back to ```tp```?

Reply:
Hey @vqd8a , at one point, setting scratch size gave you a new TeamPolicy object instead of modifying the existing one. I'm not sure if that's still the case.